src/tag-editor/utils.ts

```diff
@@ -46,3 +50,46 @@ export function useMemoizedArray<T>(array: ReadonlyArray<T>, isEqual: (prev: T,
   }, [updated]);
   return updated;
 }
+
+interface GetTagsDiffResult {
+  created: Record<string, string>;
+  removed: string[];
+}
+
+/**
+ * Compares the initial tags with the current tags passed to the tag editor
+ * and returns the differences, identifying which tags have been created or removed.
+ *
+ * This utility can be used to track tag changes and inform your tagging service about
+ * the removed and added tags.
+ *
+ * @param initialTags - The original tags fetched from the backend or tagging service.
+ * @param tags - The current tags provided to the tag editor, including any new or modified tags.
+ * @returns An object containing two arrays:
+ * - `created`: An record of tags that are new or updated (with modified values).
+ *    Each tag is represented by its `key` and `value`.
+ * - `removed`: An array of tag keys that were present in the initial tags but marked for removal.
+ */
+
+export function getTagsDiff(
+  initialTags: readonly TagEditorProps.Tag[],
+  tags: readonly TagEditorProps.Tag[]
+): GetTagsDiffResult {
+  if (initialTags.some(t => !t.existing)) {
+    warnOnce('identifyTagStates', 'all initial tags should have `existing` property set to `true`.');
+  }
+  const updated = tags.filter(tag =>
+    initialTags.some(({ key, value }) => {
+      return !tag.markedForRemoval && tag.key === key && tag.existing && tag.value !== value;
+    })
+  );
+  const created = [...tags.filter(tag => !tag.existing), ...updated]
+    .map(({ key, value }) => ({ key, value }))
+    .reduce((acc: Record<string, string>, tag) => {
+      acc[tag.key] = tag.value;
```

Comment:
Should we print a warning, if there are duplicate keys?

Reply:
Added that logic and also added it behind the `isDevelopment` flag

Reply:
`warnOnce` already has `isDevelopment` check built-in, but an extra one is fine too

Reply:
yes but I don't want to execute the checking logic in production when its not gonna warn or do anything anyways..